### PR TITLE
[ios_platform_images] Add iOS unit tests

### PR DESCRIPTION
**Pull request by jmagman**
Add `RunnerTests` XCTest iOS target to example app that creates the plugin and asserts it's not nil, proving linking is set up correctly.

ios_platform_images part of https://github.com/flutter/flutter/issues/83357

https://cirrus-ci.com/task/5061232655859712?logs=xctest#L1734
```
Test Suite 'IosPlatformImagesTests' passed at 2021-06-02 11:04:09.893.
	 Executed 1 test, with 0 failures (0 unexpected) in 0.011 (0.021) seconds
Test Suite 'RunnerTests.xctest' passed at 2021-06-02 11:04:09.897.
	 Executed 1 test, with 0 failures (0 unexpected) in 0.011 (0.032) seconds
```

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide] and the [C++, Objective-C, Java style guides]. (Note that unlike the flutter/flutter repo, the flutter/plugins repo does use `dart format`. See [plugin_tool format](../script/tool/README.md#format-code))
- [x] I signed the [CLA].
- [x] The title of the PR starts with the name of the plugin surrounded by square brackets, e.g. `[shared_preferences]`
- [x] I listed at least one issue that this PR fixes in the description above.
- [ ] I updated pubspec.yaml with an appropriate new version according to the [pub versioning philosophy].
- [x] I updated CHANGELOG.md to add a description of the change.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test exempt.
- [x] All existing and new tests are passing.

### CODE REVIEW

File: packages/ios_platform_images/example/ios/Runner.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -111,7 +163,6 @@
 				97C146FD1CF9000F007C117D /* Assets.xcassets */,
 				97C146FF1CF9000F007C117D /* LaunchScreen.storyboard */,
 				97C147021CF9000F007C117D /* Info.plist */,
-				97C146F11CF9000F007C117D /* Supporting Files */,
```
Comment (jmagman) [inline packages/ios_platform_images/example/ios/Runner.xcodeproj/project.pbxproj:114]: This group was empty.

**Review by stuartmorgan — CHANGES_REQUESTED** [review]
Same comment here as https://github.com/flutter/plugins/pull/3998#pullrequestreview-674261575; any test I can imagine for native code here can be written as a native unit test.

**Comment by jmagman** [comment]
> Same comment here as [#3998 (review)](https://github.com/flutter/plugins/pull/3998#pullrequestreview-674261575); any test I can imagine for native code here can be written as a native unit test.

Removed the target.

**Review by stuartmorgan — APPROVED** [review]
LGTM!

**Comment by stuartmorgan** [comment]
(`format` is unhappy though; that will need a minor fix)

**Pull request merged**